src/protocols/ethereum/libs/EthereumAdapter.ts

```diff
@@ -242,7 +242,7 @@ export class EthereumAdapter extends BaseProtocolAdapter {
     const web3Eth = new Web3Eth(nodeUrl);
     const nonce = await this.getTransactionCount(options.fromAccount.address!);
 
-    const hexAmount = bigIntToHex(BigInt(toWei(amount, 'ether')));
+    const hexAmount = bigIntToHex(BigInt(toWei(amount.toFixed(ETH_COIN_PRECISION), 'ether')));
```

Comment:
Is `toWei` unable to handle E notation `1e-18`?

Reply:
Yes, it throws an error that it cannot convert it